synapse/lib/stormlib/cortex.py

```diff
@@ -0,0 +1,1050 @@
+                  'args': (
+                      {'name': 'iden', 'type': 'string',
+                       'desc': 'The iden of the API to retreive.'},
+                  ),
+                  'returns': {'type': 'http:api', 'desc': 'The http:api object.'}}},
+        {'name': 'list', 'desc': 'Get all the Extneded HTTP APIs on the Cortex',
+         'type': {'type': 'function', '_funcname': 'listHttpApis', 'args': (),
+                  'returns': {'type': 'list', 'desc': 'A list of http:api objects'}}},
+        {'name': 'index', 'desc': 'Set the index for a given Extended HTTP API.',
+         'type': {'type': 'function', '_funcname': 'setHttpApiIndx',
+                  'args': (
+                      {'name': 'iden', 'type': 'string',
+                       'desc': 'The iden of the API to modify.'},
+                      {'name': 'index', 'type': 'int', 'default': 0,
+                      'desc': 'The new index of the API. Uses zero based indexing.'},
+                  ),
+                  'returns': {'type': 'int', 'desc': 'The new index location of the API.'}}},
+        {'name': 'response', 'desc': 'Make a response object. Used by API handlers automatically.',
+         'type': {'type': 'function', '_funcname': 'makeHttpResponse',
+                  'args': (
+                      {'name': 'requestinfo', 'type': 'dict',
+                       'desc': 'Request info dictionary. This is an opaque data structure which may change.'},
+                  ),
+                  'returns': {'type': 'http:api:request'}}},
+    )
+    _storm_lib_path = ('cortex', 'httpapi')
+
+    _storm_lib_perms = (
+        {'perm': ('storm', 'lib', 'cortex', 'httpapi', 'add'), 'gate': 'cortex',
+         'desc': 'Controls the ability to add a new Extended HTTP API on the Cortex.'},
+        {'perm': ('storm', 'lib', 'cortex', 'httpapi', 'get'), 'gate': 'cortex',
+         'desc': 'Controls the ability to get or list Extended HTTP APIs on the Cortex.'},
+        {'perm': ('storm', 'lib', 'cortex', 'httpapi', 'del'), 'gate': 'cortex',
+         'desc': 'Controls the ability to delete a Extended HTTP API on the Cortex.'},
+        {'perm': ('storm', 'lib', 'cortex', 'httpapi', 'set'), 'gate': 'cortex',
+         'desc': 'Controls the ability to modify a Extended HTTP API on the Cortex.'},
+    )
+
+    def getObjLocals(self):
+        return {
+            'add': self.addHttpApi,
+            'del': self.delHttpApi,
+            'get': self.getHttpApi,
+            'list': self.listHttpApis,
+            'index': self.setHttpApiIndx,
+            'response': self.makeHttpResponse,
+        }
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def makeHttpResponse(self, requestinfo):
+        requestinfo = await s_stormtypes.toprim(requestinfo)
+        return HttpReq(self.runt, requestinfo)
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def getHttpApi(self, iden):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'get'))
```

Comment:
worth checking if they have permission for that specific API or is it all or nothing?

Reply:
We can discuss that as a goback? I was leaning away from that since it does cause us to wildcard every permission.

Reply:
Per discussion omit the iden